packages/jest-reporters/src/generateEmptyCoverage.ts

```diff
@@ -27,9 +37,34 @@ export default function(
     collectCoverage: globalConfig.collectCoverage,
     collectCoverageFrom: globalConfig.collectCoverageFrom,
     collectCoverageOnlyFrom: globalConfig.collectCoverageOnlyFrom,
+    v8Coverage: globalConfig.v8Coverage,
   };
   let coverageWorkerResult: CoverageWorkerResult | null = null;
   if (shouldInstrument(filename, coverageOptions, config)) {
+    if (coverageOptions.v8Coverage) {
+      const stat = fs.statSync(filename);
```

Comment:
Any chance we can avoid this? I've noticed in the past that collecting coverage can take a long time even when almost none of the project files are covered, and I'm afraid this could contribute. The haste FS has file sizes since https://github.com/facebook/jest/pull/7580, can we somehow get that info in here?
Of course if it turns out not to be a trivial refactor we can do some profiling first to see if it'd actually be worth it.

Reply:
we don't have access to haste in here. But previously we read the file, parsed it and counted statements/branches etc. This `stat` is orders of magnitude faster than the babel approach (albeit less accurate). We don't really need the worker for v8 coverage at all, but I don't _think_ we have access to the haste map in the callee either